### ISSUE TITLE
Crash after (successful) penalty shot

### ISSUE THREAD

**Issue by schmidma** [HULKs/hulk#1401]
We crash after a penalty shot:

```
2024-07-20-HULKs-vs-SPQR/first-half/10.1.24.39/2024-07-20_14:23:06/hulk.err
```

and

```
2024-07-20-HULKs-vs-SPQR/first-half/10.1.24.39/2024-07-20_14:39:16/hulk.err
```

**Comment by knoellle** [comment]
Likely culprit in role assigment:
```rs
    if let Some(game_controller_state) = filtered_game_controller_state {
        match game_controller_state.game_phase {
            GamePhase::PenaltyShootout {
                kicking_team: Team::Hulks,
            } => return (Role::Striker, false, None),
            GamePhase::PenaltyShootout {
                kicking_team: Team::Opponent,
            } => return (Role::Keeper, false, None),
            _ => {}
        };
        if let Some(SubState::PenaltyKick) = game_controller_state.sub_state {
            return (current_role, false, None);
        }
    }
```
In penalty kick we are always enforcing the previous role which might
- leave us without a striker if there wasn't one at the time this state is entered
- cause a robot to be striker without a ball or team ball, which is fatal.

**Comment by knoellle** [comment]
Fixed by #1402
